Packs/Akamai_SIEM/Integrations/Akamai_SIEM/Akamai_SIEM.py

```diff
@@ -345,6 +372,66 @@ def get_events_command(client: Client, config_ids: str, offset: str | None = Non
         return f'{INTEGRATION_NAME} - Could not find any results for given query', {}, {}
 
 
+@logger
+def fetch_events_command(
+    client: Client,
+    fetch_time: str,
+    fetch_limit: str | int,
+    config_ids: str,
+    ctx: dict,
+) -> Iterator[Any]:
+    """Iteratively gathers events from Akamai SIEM. Stores the offset in integration context.
+
+    Args:
+        client: Client object with request
+        fetch_time: From when to fetch if first time, e.g. `3 days`
+        fetch_limit: limit of events in a fetch
+        config_ids: security configuration ids to fetch, e.g. `51000;56080`
+        ctx: The integration context
+
+    Yields:
+        (list[dict], str): events and new offset.
+    """
+    total_events_count = 0
+
+    from_epoch, _ = parse_date_range(date_range=fetch_time, date_format='%s')
+    offset = ctx.get("offset")
+    while total_events_count < int(fetch_limit):
+        events, offset = client.get_events_with_offset(config_ids, offset, FETCH_EVENTS_PAGE_SIZE, from_epoch)
+        if not events:
+            break
+        for event in events:
+            try:
+                event["_time"] = event["httpMessage"]["start"]
+                event['attackData']['rules'] = decode_message(event.get('attackData', {}).get('rules', ""))
+                event['attackData']['ruleMessages'] = decode_message(event.get('attackData', {}).get('ruleMessages', ""))
+                event['attackData']['ruleTags'] = decode_message(event.get('attackData', {}).get('ruleTags', ""))
+                event['attackData']['ruleData'] = decode_message(event.get('attackData', {}).get('ruleData', ""))
+                event['attackData']['ruleSelectors'] = decode_message(event.get('attackData', {}).get('ruleSelectors', ""))
+                event['attackData']['ruleActions'] = decode_message(event.get('attackData', {}).get('ruleActions', ""))
+                event['attackData']['ruleVersions'] = decode_message(event.get('attackData', {}).get('ruleVersions', ""))
+                event['httpMessage']['requestHeaders'] = decode_url(event.get('httpMessage', {}).get('requestHeaders', ""))
+                event['httpMessage']['responseHeaders'] = decode_url(event.get('httpMessage', {}).get('responseHeaders', ""))
+            except Exception as e:
+                config_id = event.get('attackData', {}).get('configId', "")
+                policy_id = event.get('attackData', {}).get('policyId', "")
+                demisto.debug(f"Couldn't decode event with {config_id=} and {policy_id=}, reason: {e}")
+
+        total_events_count += len(events)
+        yield events, offset
+
+
+def decode_url(headers):
+    decoded_lines = urllib.parse.unquote(headers).split("\r\n")
```

Comment:
are we sure always has \r\n ? Is it possible the message doesnt come from a windows system? check with @dmitryzinger but i think might be better to split on \n and drop \r